### PR TITLE
Adding a new summon feature to call someone

### PR DESCRIPTION
**Pull request by ghoranyi**
With this PR, we are adding a new feature (and a new module) to Pasha: if you type 'pasha summon name message', then it will look up the phone number of the person (from PagerDuty) and then makes a phone call and sends an SMS to the user. Not necessary to specify the full name, as it works the same way as the whois command (fraction of the name or mention name can be used). This functionality won't work if the PagerDuty module is not available.
The module will recognise if a phone number is passed instead of a name, and then it will skip the PD lookup part. This works event when the PD module is not loaded.
We also extended the functionality of the whois command, as it will give back the user's phone number(s) if the PagerDuty module is available.

### CODE REVIEW

**Comment by zsol** [comment]
Awesome PR

File: run_scripts/run_bash_adapter.sh [hunk 1]
```diff
@@ -29,7 +29,11 @@ PROVISION_PORT=
 
 PRIO1_PLAYBOOK_URL=
 PRIO1_MONITORED_WEBSITE=
-PRIO1_ROOM=Ops
+PRIO1_ROOM=Ooops
```
Comment (csizsek) [inline run_scripts/run_bash_adapter.sh]: this should go to the config file

File: scripts/pasha_main.coffee [hunk 2]
```diff
@@ -127,17 +129,19 @@ module.exports = (robot) ->
             "#{botName} status: display or set prio1 status, " +
             "see '#{botName} status help' for details"
         robot.receive(new TextMessage(msg.message.user,
-            "#{botName} changelog helpFromMain"))
+            "#{botName} changelog help_from_main"))
         robot.receive(new TextMessage(msg.message.user,
```
Comment (csizsek) [inline scripts/pasha_main.coffee:133]: these kind of errors are already fixed in master and they don't belong to this feature anyways

File: scripts/pasha_pagerduty.coffee [hunk 3]
```diff
@@ -200,11 +294,20 @@ module.exports = (robot) ->
         response = "#{botName} alert trigger <service_name> <description>: " +
             "triggers an alert to the service with the specified name\n" +
             "#{botName} alert list: " +
-            "lists the details of the active Pagerduty alerts"
+            "lists the details of the active Pagerduty alerts\n" +
+            "#{botName} alert phone <email>: " +
+            "return the phone number of a user\n"
```
Comment (csizsek) [inline scripts/pasha_pagerduty.coffee:299]: the command name here is misleading, for a command like this i would expect that pasha actually does alert the person

File: scripts/pasha_twilio.coffee [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+client = require('twilio')(twilioAccountSid, twilioAuthToken)
+
+# Helpers
+standardizePhoneNumber = (number) ->
+  n = number.replace(/\+/g, "").replace(/\ /g, "").replace(/\(/g, "").replace(/\)/g, "").replace(/-/g, "")
+  return "+#{n}"
+
+sendSms = (number, reason, roomName, msg, name) ->
+
+  smsReason = "This is an automated text message from Prezi Pasha. " +
```
Comment (csizsek) [inline scripts/pasha_twilio.coffee]: Why is 'Prezi' hardcoded?

Comment (csizsek) [inline scripts/pasha_twilio.coffee]: and 'Pasha'?

File: scripts/pasha_twilio.coffee [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+  }
+  smsCallback = (err, message) ->
+     scribeLog "twilio response Sid: #{message.sid}"
+  client.messages.create(smsPayload, smsCallback)
+
+phoneCall = (number, reason, roomName, msg, name) ->
+  encodedRoomName = encodeURIComponent(roomName)
+  encodedReason = encodeURIComponent(reason)
+  twimletUrl = "http://twimlets.com/message?Message%5B0%5D=" +
+    "This%20is%20an%20automated%20phone%20call%20from%20Prezi%20Pasha." +
```
Comment (csizsek) [inline scripts/pasha_twilio.coffee]: Why is 'Prezi' and 'Pasha' hardcoded?

Comment (csizsek) [inline scripts/pasha_twilio.coffee]: TODO: ask users about the notification text, maybe it could be shorter

File: scripts/pasha_twilio.coffee [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+        wasPickedUp = "not "
+      callStatus = "the call to #{name} (#{number}) was #{wasPickedUp}picked up"
+      msg.reply callStatus
+      scribeLog callStatus
+    )
+    clearInterval callStatusCallbackId
+  callStatusCallbackId = setInterval(callStatusCallback, 1000 * 90)
+
+# Commands
+summonByPhoneNumber = /summon (\+?[0-9\ \-\(\)]*) (.+)$/i
```
Comment (csizsek) [inline scripts/pasha_twilio.coffee]: do we have tests for these?

File: scripts/pasha_twilio.coffee [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+summonByName = /summon ([^\+^0-9^\ ^\(^\)]*) (.+)$/i
```
Comment (csizsek) [inline scripts/pasha_twilio.coffee]: regexes?

File: scripts/pasha_twilio.coffee [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+            u = util.getUser(name, msg.message.user.name, pashaState.users)
+            if u
+              pagerduty.phone(u.email, (numbers) ->
+                  for number in numbers
+                      phoneNumber = standardizePhoneNumber(number)
+
+                      sendSms(phoneNumber, reason, roomName, msg, name)
+                      scribeLog "sent SMS to: #{name} (#{phoneNumber})"
+                      msg.reply "sent SMS to: #{name} (#{phoneNumber})"
+
```
Comment (csizsek) [inline scripts/pasha_twilio.coffee]: DRY: this is code duplication with line #152

File: test/pasha_twilio_test.coffee [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+    it 'should register twilio commands in robot.registeredCommands', () ->
+        for command, regexes of pashaTwilioCommands
+            assert.property(robot.registeredCommands, command)
+            regexes = regexes.toString()
+            registeredRegexes = robot.registeredCommands[command].toString()
+            assert.equal(regexes, registeredRegexes)
+
+
+    it 'should not summon by name if no pagerduty module is available', (done) ->
+        rewire = require('rewire')
```
Comment (csizsek) [inline test/pasha_twilio_test.coffee]: is this needed in this test?

File: test/pasha_twilio_test.coffee [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+        twilioAuthToken = constant.twilioAuthToken
+        twilio = require('twilio')(twilioAccountSid, twilioAuthToken)
+        mockMessages = sinon.mock(twilio.messages)
+        mockMessages.create = sinon.spy()
+        mockCall = sinon.spy()
+        pashaTwilio.__set__('client.messages', mockMessages)
+        pashaTwilio.__set__('client.makeCall', mockCall)
+        adapter.on 'reply', (envelope, response) ->
+            sinon.assert.calledOnce(mockMessages.create)
+            smsReason = "This is an automated text message from Prezi Pasha. " +
```
Comment (csizsek) [inline test/pasha_twilio_test.coffee]: don't hardcode 'prezi' or 'pasha'

File: scripts/pasha_twilio.coffee [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+     body: smsReason,
+  }
+  smsCallback = (err, message) ->
+     scribeLog "twilio response Sid: #{message.sid}"
+  client.messages.create(smsPayload, smsCallback)
+
+phoneCall = (number, reason, roomName, msg, name) ->
+  encodedRoomName = encodeURIComponent(roomName)
+  encodedReason = encodeURIComponent(reason)
+  twimletUrl = "http://twimlets.com/message?Message%5B0%5D=" +
```
Comment (nyuwec) [inline scripts/pasha_twilio.coffee]: Isn't it easier to encode the whole, concatenated string?
In that case the message could be edited easily, or could be localized

**Pull request merged**